tests/testthat/test-survival-tune_race_anova.R

```diff
@@ -241,6 +276,45 @@ test_that("race tuning (anova) survival models with integrated metric", {
   expect_equal(metric_aov_all[0,], exp_metric_all)
   expect_true(all(metric_aov_all$.metric == "brier_survival_integrated"))
 
+  # test prediction collection -------------------------------------------------
+
+  integrated_ptype <-
+    structure(
+      list(.pred = list(), id = character(0), .row = integer(0),
+           cost_complexity = numeric(0),
+           event_time = structure(numeric(0), type = "right", dim = c(0L, 2L),
+                                  dimnames = list(NULL, c("time", "status")),
+                                  class = "Surv"),
+           .config = character(0)), row.names = integer(0),
+      class = c("tbl_df", "tbl", "data.frame"))
+
+  integrated_list_ptype <-
+    tibble::tibble(
+      .eval_time = numeric(0),
+      .pred_survival = numeric(0),
+      .weight_censored = numeric(0)
+    )
+
+  integrated_oob <-
+    aov_integrated_res %>%
+    mutate(oob = map_int(splits, ~ nrow(assessment(.x)))) %>%
+    pluck("oob") %>%
+    sum()
+
+  unsum_pred <- collect_predictions(aov_integrated_res)
+  expect_equal(unsum_pred[0,], integrated_ptype)
+  expect_equal(nrow(unsum_pred), integrated_oob * nrow(aov_finished))
+
+  expect_equal(unsum_pred$.pred[[1]][0,], integrated_list_ptype)
+  expect_equal(nrow(unsum_pred$.pred[[1]]), length(time_points))
+
+
+  sum_pred <- collect_predictions(aov_integrated_res, summarize = TRUE)
+  expect_equal(sum_pred[0,], integrated_ptype[, names(integrated_ptype) != "id"])
+  expect_equal(nrow(sum_pred), 728)
```

Comment:
where does the `728` come from? I'm struggling to reproduce this locally.

Reply:
FYI also `expect_equal(nrow(sum_pred), nrow(sim_tr) * nrow(aov_finished))` fails for this one locally.

Reply:
So I figured it out. I did a low number of resamples (6) to save computing time. With this small number, not all of the training set points are in at least one of the OOB assessment sets.  When you multiply this by the number of parameters that finish the race, you get the right number. I'll update the logic. 

```
> num_hold_out_rows <- 
+     sim_rs %>% 
+     mutate(
+         oob_rows = map(splits, ~ as.integer(.x, data = "assessment"))
+     )  %>% 
+     pluck("oob_rows") %>% 
+     unlist() %>% 
+     unique() %>% 
+     length()
> num_hold_out_rows
[1] 364
> nrow(sim_tr)
[1] 375
> num_hold_out_rows * nrow(aov_finished)
[1] 728
```